### ISSUE TITLE
Should not claim to expect `>` for "error: expected one of ..., `>`, ... or an operator, found `>`"

### ISSUE THREAD

**Issue by insaneinside** [rust-lang/rust#24780]
[Test-case playpen.](http://is.gd/khyi4o)

This currently happens for both beta and nightly.

**Comment by tbelaire** [comment]
Just inlining the example and error:
```
fn foo() -> Vec<usize>> {
    //                ^
    // error: expected one of `(`, `+`, `::`, `<`, `>`, `where`, or `{`, found `>`
    Vec::new()
}
```

**Comment by bltavares** [comment]
This issue has been resolved after #31455 got merged.

```
rustc 1.8.0-nightly (3f4227af1 2016-02-10)
```

The error now is the following:
```
tmp/24780.rs:1:23: 1:24 error: expected one of `!`, `::`, `where`, or `{`, found `>`
tmp/24780.rs:1 fn foo() -> Vec<usize>> {
                                     ^
error: aborting due to previous error
```

**Comment by alexcrichton** [comment]
Thanks @bltavares!
